devito/symbolics/manipulation.py

```diff
@@ -128,7 +128,10 @@ def _(mapper, rule):
 
 @singledispatch
 def _uxreplace_handle(expr, args, kwargs):
-    return expr.func(*args)
+    try:
```

Comment:
why?

Reply:
You end up here wit hsome nexted methods such as `Abs`  that need not be evaluated. For example this leads to undoing `pow_to_mul` for `norm` with some language because the `Abs(a*a)` gets evaluated and re-simplified as `Abs(**2)`. So basically any sympy "evaluable"  needs to be not evaluated here.

Reply:
maybe the pow_to_mul output should enforced to be of type `Unevaluated*`, https://github.com/devitocodes/devito/blob/main/devito/symbolics/unevaluation.py

Reply:
That'd still be just a local hack it doesn't ensure it will not evaluate generic sympy expression we didn't think of here

Reply:
I'm not suggesting as an alternative...

But freezing the exprs converted by pow_to_mul would _not_ be a hack